### ISSUE TITLE
Change commentPattern option of no-fallthrough to be case insensitive

### ISSUE THREAD

**Issue by Standard8** [eslint/eslint#12074]
**What rule do you want to change?**

`no-fallthrough` - change the `commentPattern` regexp parameter to be treated as case insensitive.

**Does this change cause the rule to produce more or fewer warnings?**

Fewer.

**How will the change be implemented? (New option, new default behavior, etc.)?**

Adjust the existing option.

**Please provide some example code that this change will affect:**

```js
switch(foo1) { case 0: a(); /* no break */ case 1: b(); }
switch(foo2) { case 0: a(); /* NO BREAK */ case 1: b(); }
```
ESLint Rule:
```
"no-fallthrough": [
  "error",
  {
    commentPattern: "no break"
  }
]
```

**What does the rule currently do for this code?** 

The `foo1` switch statement passes, `foo2` fails.

**What will the rule do after it's changed?**

Both `foo1` and `foo2` will pass.

**Are you willing to submit a pull request to implement this change?** Yes

I'm not sure if this counts as a bug or a breaking change, hence filing as an issue to begin with.

**Comment by Standard8** [comment]
Note: the default regular expression `/falls?\s?through/iu` is already treated as case-insensitive, which is why I think it makes sense to make this change.

**Comment by kaicataldo** [comment]
This would be a breaking change if we enabled this behavior by default. I do think it makes sense to allow for this behavior, but I'm not sure what the best way to go about it is.

Some options:
1. Put the behavior suggested here behind a flag (maybe `caseInsensitive: boolean`?)
1. Allow for configuration to pass an actual RegExp to the configuration (though this would only work in .js configuration files)
1. Allow for configuration to pass a full RegExp as a string (i.e. `commentPattern: "/no\s*break/i" )

**Comment by platinumazure** [comment]
I would vote for option 1 for this issue, and I think we should have a larger issue about different ways to support regex options (including standardizing on something across the core rules if possible).

**Comment by g-plane** [comment]
If we apply option 3, it'll be better that the comment patterns can be an array, which allows user to define different patterns instead of a single string.
